server/util/scratchspace/scratchspace.go

```diff
@@ -0,0 +1,54 @@
+package scratchspace
+
+import (
+	"flag"
+	"os"
+	"path/filepath"
+
+	"github.com/buildbuddy-io/buildbuddy/server/util/status"
+)
+
+var (
+	tempDir_doNotUseDirectly = flag.String("storage.tempdir", filepath.Join(os.TempDir(), "buildbuddy-scratch"), "Staging directory for ephemeral files. If unspecified, defaults to 'TEMPDIR/buildbuddy' where TEMPDIR is the OS-specific temp dir.")
+)
+
+// Init creates the scratch directory if needed and deletes any pre-existing
+// files. Calling this function is required before using any other functions in
+// this package.
+func Init() error {
+	if err := os.RemoveAll(TempDir()); err != nil {
+		return status.InvalidArgumentErrorf("failed to clear scratch directory: %s", err)
+	}
+	if err := os.MkdirAll(TempDir(), 0755); err != nil {
+		return status.InvalidArgumentErrorf("failed to create scratch directory: %s", err)
+	}
+	return nil
+}
+
+// TempDir returns the directory where scratch files are written.
+func TempDir() string {
```

Comment:
nit: consider making this private if it's not used outside of this pcakage

Reply:
done